src/core/server/http/http_config.ts

```diff
@@ -195,7 +195,13 @@ export class HttpConfig {
     rawExternalUrlConfig: ExternalUrlConfig
   ) {
     this.autoListen = rawHttpConfig.autoListen;
-    this.host = rawHttpConfig.host;
+    // TODO: Consider dropping support for '0' in v8.0.0. This value is passed
+    // to hapi, which validates it. Prior to hapi v20, '0' was considered a
+    // valid host, however the validation logic internally in hapi was
+    // re-written for v20 and hapi no longer considers '0' a valid host. For
+    // details, see:
+    // https://github.com/elastic/kibana/issues/86716#issuecomment-749623781
+    this.host = rawHttpConfig.host === '0' ? '0.0.0.0' : rawHttpConfig.host;
```

Comment:
@watson We should probably follow up in all the places where we use this for testing, replace it with `0.0.0.0` and then remove it from 0?

Thanks for the fix BTW

Reply:
I think it's best to support this in 7.x and then wait till 8.0.0 to remove it

Reply:
Agree. But the 8.0.0 removal could already happen now. An email could be sent to the teams to give a heads up that it will be removed in `x` weeks with the fix that needs to happen.

Reply:
Fine with me 😃 I just wasn't sure what our policy on diverging the branches was. I've heard some say that we want to keep `master` and `7.x` in sync as much as possible to ease backporting. I presume that would mean that we hold off diverging the branches until closer to the `8.0.0` release (hence why we have the big todo-list of things we want to break).

On the other and, it's easier to get it in now when it's all fresh in our heads. I'm personally leaning towards the latter, so I'll create a PR to remove this in `master` and then people can object if they want in the PR.

Reply:
No strong opinion, I let the KB team comment ;-)

Reply:
I've created #87114 - You mentioned that the docs use `0` as the hostname. Where is that?

Reply:
I did mention docs? :-)

Reply:
Strike that... It was @graphaelli